ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1377,12 +1379,14 @@ func (r *RayClusterReconciler) reconcileAutoscalerServiceAccount(ctx context.Con
 				"If users specify ServiceAccountName for the head Pod, they need to create a ServiceAccount themselves. "+
 					"However, ServiceAccount %s is not found. Please create one. "+
 					"See the PR description of https://github.com/ray-project/kuberay/pull/1128 for more details.", namespacedName.Name), "ServiceAccount", namespacedName)
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "Failed", "Failed to reconcile RayCluster %s/%s due to %s", instance.Namespace, instance.Name, err)
```

Comment:
Is there a helpful message in the error telling users they should create the service account themselves?

Reply:
currently we only see something like this
```
Events:
  Type     Reason  Age                   From                   Message
  ----     ------  ----                  ----                   -------
  Warning  Failed  97s (x16 over 4m21s)  raycluster-controller  Failed to reconcile RayCluster default/raycluster-sample due to ServiceAccount "random-sa" not found
```
do you think we should propagate entire msg
https://github.com/ray-project/kuberay/blob/146d5ea57a4c4b6729cc5aa98c84faac5c36bd5d/ray-operator/controllers/ray/raycluster_controller.go#L1378-L1381
to the event message or we should just add the err msg with "Please create your service account %s in namespace %s beforehand"

Reply:
Yes, I think having a clear message that users should create the account by themselves can save both users and us a lot of time in the future.